dialect/sql/schema/atlas.go

```diff
@@ -6,23 +6,230 @@ package schema
 
 import (
 	"context"
+	"crypto/md5"
 	"database/sql"
 	"errors"
 	"fmt"
-	"io/fs"
-	"io/ioutil"
+	"net/url"
 	"sort"
 	"strings"
 
 	"ariga.io/atlas/sql/migrate"
 	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlclient"
 	"ariga.io/atlas/sql/sqltool"
-
 	"entgo.io/ent/dialect"
 	entsql "entgo.io/ent/dialect/sql"
 	"entgo.io/ent/schema/field"
 )
 
+// Atlas atlas migration engine.
+type Atlas struct {
+	atDriver   migrate.Driver
+	sqlDialect sqlDialect
+
+	legacy      bool // if the legacy migration engine instead of Atlas should be used
+	withFixture bool // deprecated: with fks rename fixture
+	sum         bool // deprecated: sum file generation will be required
+
+	universalID     bool // global unique ids
+	dropColumns     bool // drop deleted columns
+	dropIndexes     bool // drop deleted indexes
+	withForeignKeys bool // with foreign keys
+	mode            Mode
+	hooks           []Hook            // hooks to apply before creation
+	diffHooks       []DiffHook        // diff hooks to run when diffing current and desired
+	applyHook       []ApplyHook       // apply hooks to run when applying the plan
+	skip            ChangeKind        // what changes to skip and not apply
+	dir             migrate.Dir       // the migration directory to read from
+	fmt             migrate.Formatter // how to format the plan into migration files
+
+	driver  dialect.Driver // driver passed in when not using an atlas URL
+	url     *url.URL       // url of database connection
+	dialect string         // Ent dialect to use when generating migration files
+
+	types []string // pre-existing pk range allocation for global unique id
+}
+
+// Diff compares the state read from a database connection or migration directory with the state defined by the Ent
+// schema. Changes will be written to new migration files.
+func Diff(ctx context.Context, u, name string, tables []*Table, opts ...MigrateOption) (err error) {
+	m, err := NewMigrateURL(u, opts...)
+	if err != nil {
+		return err
+	}
+	return m.NamedDiff(ctx, name, tables...)
+}
+
+// NewMigrate creates a new Atlas form the given dialect.Driver.
+func NewMigrate(drv dialect.Driver, opts ...MigrateOption) (*Atlas, error) {
+	a := &Atlas{driver: drv, withForeignKeys: true, mode: ModeInspect, sum: true}
+	for _, opt := range opts {
+		opt(a)
+	}
+	a.dialect = a.driver.Dialect()
+	if err := a.init(); err != nil {
+		return nil, err
+	}
+	return a, nil
+}
+
+// NewMigrateURL create a new Atlas from the given url.
+func NewMigrateURL(u string, opts ...MigrateOption) (*Atlas, error) {
+	parsed, err := url.Parse(u)
+	if err != nil {
+		return nil, err
+	}
+	a := &Atlas{url: parsed, withForeignKeys: true, mode: ModeInspect, sum: true}
```

Comment:
ModeInspect? No ModeReplay as default?

Reply:
Current behavior is inspection. I want to give users time to try replay and switch to it as default in a future release